apps/meteor/client/hooks/omnichannel/useOmnichannelContact.ts

```diff
@@ -0,0 +1,43 @@
+import { ILivechatVisitor } from '@rocket.chat/core-typings';
+import { useEndpoint } from '@rocket.chat/ui-contexts';
+import { useQuery } from '@tanstack/react-query';
+import { useState } from 'react';
+
+import { parseOutboundPhoneNumber } from '../../../ee/client/lib/voip/parseOutboundPhoneNumber';
+
+type Contact = {
+	name: string;
+	phone: string;
+};
+
+type ContactQuery = {
+	isLoading: boolean;
+	isError: boolean;
+	contact: Contact;
+};
+
+const createContact = (phone: string, data: Pick<ILivechatVisitor, 'name'> | null): Contact => ({
+	phone,
+	name: data?.name || '',
+});
+
+export const useOmnichannelContact = (ogPhone: string, name = ''): ContactQuery => {
+	const getContactBy = useEndpoint('GET', '/v1/omnichannel/contact.search');
+	const phone = parseOutboundPhoneNumber(ogPhone);
+	const [defaultContact] = useState<Contact>({ phone, name });
+
+	const {
+		data: contact,
+		isLoading,
+		isError,
+	} = useQuery(['getContactsByPhone', phone], async (): Promise<Contact> => {
+		const { contact } = await getContactBy({ phone });
+		return createContact(phone, contact);
+	});
+
+	return {
+		contact: contact || defaultContact,
+		isLoading,
+		isError,
```

Comment:
c'mon i asked you to return the query to actually use it

Reply:
> I'm going to ask you to pay more attention when creating custom hooks, ok it's custom, but I really hate it when a query becomes a default data and the application no longer knows if it's loading or if the data is undefined.
> 
> however this is an alteration that does not harm anyone. But avoid spreading it around.

I don't seem to understand what you expect from this return. If you're available i'd love to talk and get to know what's on your mind.